From 67421ea36e27fffe2ea1bb89d3f72e44d2212ea5 Mon Sep 17 00:00:00 2001
From: Imre Kaloz <kaloz@openwrt.org>
Date: Thu, 16 Mar 2023 19:26:48 +0530
Subject: [PATCH 113/500] OpenWrt: 230-openwrt_lzma_options.patch

use the openwrt lzma options for now

lede-commit: 548de949f392049420a6a1feeef118b30ab8ea8c
Signed-off-by: Imre Kaloz <kaloz@openwrt.org>

Change-Id: I702b9b616728935da758c539d1c6aaa8388915ac
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 7381e93670555e92d0cc041f62fb34b3de171377)
---
 lib/decompress.c     | 1 +
 scripts/Makefile.lib | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/decompress.c b/lib/decompress.c
index ab3fc90ffc64..b9c1e51e4ebb 100644
--- a/lib/decompress.c
+++ b/lib/decompress.c
@@ -53,6 +53,7 @@ static const struct compress_format compressed_formats[] __initconst = {
 	{ {0x1f, 0x9e}, "gzip", gunzip },
 	{ {0x42, 0x5a}, "bzip2", bunzip2 },
 	{ {0x5d, 0x00}, "lzma", unlzma },
+	{ {0x6d, 0x00}, "lzma-openwrt", unlzma },
 	{ {0xfd, 0x37}, "xz", unxz },
 	{ {0x89, 0x4c}, "lzo", unlzo },
 	{ {0x02, 0x21}, "lz4", unlz4 },
diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 68d0134bdbf9..40db5c0f1a95 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -456,10 +456,10 @@ quiet_cmd_bzip2_with_size = BZIP2   $@
 # ---------------------------------------------------------------------------
 
 quiet_cmd_lzma = LZMA    $@
-      cmd_lzma = cat $(real-prereqs) | $(LZMA) -9 > $@
+      cmd_lzma = cat $(real-prereqs) | $(LZMA) e -d20 -lc1 -lp2 -pb2 -eos -si -so > $@
 
 quiet_cmd_lzma_with_size = LZMA    $@
-      cmd_lzma_with_size = { cat $(real-prereqs) | $(LZMA) -9; $(size_append); } > $@
+      cmd_lzma_with_size = { cat $(real-prereqs) | $(LZMA) e -d20 -lc1 -lp2 -pb2 -eos -si -so; $(size_append); } > $@
 
 quiet_cmd_lzo = LZO     $@
       cmd_lzo = cat $(real-prereqs) | $(KLZOP) -9 > $@
-- 
2.34.1

